spec/models/user_spec.rb

```diff
@@ -176,24 +176,77 @@
     end
   end
 
-  describe '.search' do
-    subject { User.search(params) }
+  describe '.filter' do
     let(:params) { {} }
 
-    before(:each) do
-      FactoryGirl.create(:user, first_name: 'Bob', created_at: 5.days.ago)
-      FactoryGirl.create(:user, first_name: 'Marley', created_at: 2.days.ago)
-      FactoryGirl.create(:user, first_name: 'Janice', display_profile: false)
+    context 'has filters' do
+      before(:each) do
+        @user1 = FactoryGirl.create(:user, longitude: 19.15)
+        @user2 = FactoryGirl.create(:user, longitude: 27.74)
+        @project = FactoryGirl.create(:project)
+      end
+
+      it 'filters users for project' do
+        @user1.follow @project
+        @user2.stop_following @project
+        params['project_filter'] = @project.id
+
+        results = User.filter(params).allow_to_display
+        
+        expect(results).to include(@user1)
+        expect(results).not_to include(@user2)
```

Comment:
you could use the `match_array` matcher for these expectations, that way you will only need to write one `expect`